doc/manual/src/command-ref/nix-instantiate.md

```diff
@@ -35,13 +35,50 @@ standard input.
 
   - `--parse`\
     Just parse the input files, and print their abstract syntax trees on
-    standard output in ATerm format.
+    standard output as a Nix expression.
 
   - `--eval`\
     Just parse and evaluate the input files, and print the resulting
     values on standard output. No instantiation of store derivations
     takes place.
 
+    > **Warning**
+    >
+    > This option produces ambiguous output which is not suitable for machine
+    > consumption. For example, these two Nix expressions print the same result
+    > despite having different types:
```

Comment:
Arguably they have the same type at this time, `tThunk`, maybe “despite not being equal” would be better? In any case, this is sort of the point of not forcing the value completely, `nix eval` will also print the same output for non-equal values/values where the type isn't the same.

Wouldn't be clearer to just say that the problem with the output is that it produces something that can be parsed as a Nix expression, but would not return the same result?

Reply:
I agree and actually it is ambiguous what the ambiguity is. It can be either
 - intentional ambiguity because `--strict` wasn't specified,
 - or ambiguity in the mind of the reader, who might "naively" (can't blame) read it is an expression instead of a mere value, while non-value expressions are not printed by this command.

<details><summary>rant: why lawyering about "non-value expression"</summary>

I don't think we should ever conflate values and their _expression literal_ counterparts, but someone might take that view and I had to be explicit. "expression" should have been enough IMO. The thing represented by `Expr`.
We really need to define some language terminology in the manual, but I do feel that we should just improve this text first.

</details>

Reply:
Fixed in #9755.